content/How_To/floating_origin/floating_origin.md

```diff
@@ -0,0 +1,126 @@
+
+This does not mean that the camera cannot move, though! It "moves", but not directly. Here is
+where lies the trick; instead of changing the real camera position, we use a separate, double precision
+floating-point Vector3 which stores the camera position. The real camera position is kept always (0, 0, 0).
+
+We do the same for the object; all of them get a separate, double precision Vector3 to store their coordinates.
+We don't set their real position directly; instead, we also set their separate coordinates.
+
+Then, on a loop which happens every frame just before rendering, we subtract the double camera position
+from each object's double position, and that difference is copied to that object's real position.
+
+The result is that the camera is indeed kept always at origin, and the objects float around, removing
+huge coordinates from the objects that are close to the camera. That is, imprecision only happens very far
+from the camera, and as they are very far anyway, we cannot see the jittering. =)
+
+![Pic02](/img/how_to/floating_origin/pic02.jpg)
+
+Let's use an example:
+
+On a solar system, we have an asteroid located at (10000000, 0, 10000000).
+Our camera is close to the asteroid, at (10000000, 0, 10000500).
+
+Normally, that would certainly cause jittering, because both objects are very far from the world's origin. But with
+our trick, the jittering does not happen -- remember, we subtract camera double position from the object double position
+and then set the object real position at that offset, keeping the camera always at (0, 0, 0):
+
+Object Position: (10000000, 0, 10000000)  
+Camera Position: (10000000, 0, 10000500)  
+Offset: (0, 0, -500)
+
+The offset is small enough to be absolutely precise even with only 32 bits floating-point. The GPU is very happy with that
+and we don't see any jittering.
+
+We just need to set the object at (0, 0, -500) and it will have the same visual effect as if we used their real coordinates,
+but with no jittering.
+
+## FLOATING-ORIGIN EXAMPLE
+
+You can find a working example here:
+https://imerso.github.io/babylon.js/floating_origin
+
+The source-code for the example is here:
+https://github.com/imerso/bjs_floating_origin
+
+If you decide to use floating-origin, all your objects will have to use the same trick,
+or your game won't work properly. You will need to create one instance of OriginCamera
+and at least one instance of Entity.
+
+OriginCamera is a special camera which has a separate position control stored as doublepos (and its target as doubletgt). 
+You must stop using position and target from camera, and use their double precision counterparts doublepos and doubletgt.
+
+All objects from scene must then be parented to a Entity instance. Entity also has doublepos property which is double precision coordinate. 
+You must use its doublepos to set object position instead of position directly.
+
+So, let's say that we want a sphere with double precision:
+
+```  
```

Comment:
Adding "javascript" after those will make the code format-able in IDEs and will allow the MD parser to work better (i.e. language-specific)

```suggestion
```javascript
```

Reply:
Probably in the wrong place, also there are links to my github pages which is probably wrong as well.

Not sure what would be the best to do, maybe those two little classes could be a package hosted somewhere else and the demo made simpler.

If you instruct me what is the best I'll do the required changes. Thanks.